packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpType.cs

```diff
@@ -155,23 +154,22 @@ internal CSharpType(TypeProvider implementation, bool isValueType = false, bool
             _declaringType = declaringType;
             _arguments = args ?? Array.Empty<CSharpType>();
             _isPublic = isPublic;
-            _unionItemTypes ??= Array.Empty<CSharpType>();
         }
 
         public string Namespace { get { return _namespace; } }
         public string Name { get { return _name; } }
         public CSharpType? DeclaringType { get { return _declaringType; } }
         public bool IsValueType { get { return _isValueType; } }
         public bool IsEnum { get { return _isEnum; } }
-        public bool IsLiteral => Literal is not null;
-        public bool IsUnion => _isUnion ??= UnionItemTypes.Count > 0;
+        public bool IsLiteral => _literal is not null;
+        public bool IsUnion => _unionItemTypes?.Count > 0;
         public bool IsPublic { get { return _isPublic; } }
         public bool IsFrameworkType => _type != null;
         public bool IsNullable { get { return _isNullable; } }
         public bool IsGenericType => Arguments.Count > 0;
         public bool IsCollection => _isCollection ??= TypeIsCollection();
         public Type FrameworkType => _type ?? throw new InvalidOperationException("Not a framework type");
-        public Constant? Literal { get; private init; }
+        public object Literal => _literal ?? throw new InvalidOperationException("Not a literal type");
```

Comment:
I change this to let this work in the same way as "FrameworkType" and "Implementation", we will have to check if this is a literal type by calling `IsLiteral` before we could use `Literal`.
This will avoid the usage of:
```
type.IsLiteral && type.Literal is not null
```
because `type.IsLiteral` should imply `type.Literal` is not null, we should never need to check them both.